code/modules/chemistry/tools/extinguisher.dm

```diff
@@ -1,3 +1,20 @@
+var/global/list/extinguisher_blacklist_clog = list("vomit",
+	"blackpowder",
+	"blood",
+	"ketchup",
+	"gvomit",
+	"carbon",
+	"cryostylane",
+	"chickensoup",
+	"salt")
+var/global/list/extinguisher_blacklist_melt = list("acid",
+	"pacid",
+	"phlogiston",
+	"big_bang",
+	"clacid",
+	"nitric_acid",
+	"firedust")
+
 /obj/item/extinguisher
 	name = "fire extinguisher"
 	icon = 'icons/obj/items/items.dmi'
```

Comment:
notably, `foof` and `infernite` are missing from this list because they're specialcased in extinguisher code.
Should also block them from the sprayer